File: app/src/test/java/com/dampcake/robotest/MainActivityTest.java
package com.dampcake.robotest;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricGradleTestRunner;
import org.robolectric.annotation.Config;
import org.robolectric.shadows.ShadowToast;

import static org.junit.Assert.assertEquals;

@RunWith(RobolectricGradleTestRunner.class)
@Config(constants = BuildConfig.class, sdk = 21)
public class MainActivityTest {

    @Test
    public void testToastDisplayed() {
        Robolectric.setupActivity(MainActivity.class);

        assertEquals("Show Toast", ShadowToast.getTextOfLatestToast());
    }
}

File: app/src/test/java/com/dampcake/robotest/MainActivityTest.kt
package com.dampcake.robotest

import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.Robolectric
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config
import org.robolectric.shadows.ShadowToast

@RunWith(RobolectricTestRunner::class)
@Config(constants = BuildConfig::class, sdk = [21])
class MainActivityTest {
    @Test
    fun testToastDisplayed() {
        Robolectric.setupActivity(MainActivity::class.java)
        Assert.assertEquals("Show Toast", ShadowToast.getTextOfLatestToast())
    }
}